frontend/src/components/elements/Metric/styled-components.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import styled from "@emotion/styled"
+
+export const StyledMetricSpan = styled.span(({ theme }) => ({
+  marginRight: theme.spacing.twoXS,
+}))
+
+export const StyledMetricText = styled.div(({ theme }) => ({
+  overflowWrap: "normal",
+  textOverflow: "ellipsis",
+  width: "100%",
+  overflow: "hidden",
+  whiteSpace: "nowrap",
+  fontFamily: theme.genericFonts.bodyFont,
+}))
```

Comment:
My only request will likely be perhaps calling it `StyledTruncateText`....mainly because all of these are Metric Text...the purpose of this is to truncate the text.